src/analysis_file.py

```diff
@@ -7,207 +7,92 @@
-            if plan.raw_field not in demog_keys:
-                demog_keys.append(plan.raw_field)
-
-        # Convert demog codes to their string values
-        for td in data:
-            td.append_data(
-                {plan.analysis_file_key: plan.code_scheme.get_code_with_id(td[plan.coded_field]["CodeID"]).string_value
-                 for plan in PipelineConfiguration.DEMOG_CODING_PLANS
-                 if plan.analysis_file_key is not None},
-                Metadata(user, Metadata.get_call_location(), time.time())
-            )
-
-        # Convert the operator code to its string value
-        for td in data:
-            td.append_data(
-                {"operator": CodeSchemes.SOMALIA_OPERATOR.get_code_with_id(
-                    td["operator_coded"]["CodeID"]).string_value},
-                Metadata(user, Metadata.get_call_location(), time.time())
-            )
-
-        # Convert RQA and follow up binary codes to their string values
-        for td in data:
-            td.append_data(
-                {plan.binary_analysis_file_key:
-                     plan.binary_code_scheme.get_code_with_id(td[plan.binary_coded_field]["CodeID"]).string_value
-                 for plan in PipelineConfiguration.RQA_CODING_PLANS + PipelineConfiguration.FOLLOW_UP_CODING_PLANS if
-                 plan.binary_code_scheme is not None},
-                Metadata(user, Metadata.get_call_location(), time.time())
-            )
-
-        # Translate the RQA and follow up survey codes to matrix values
+        # Set the list of keys to be exported and how they are to be handled when folding
+        export_keys = ["uid", consent_withdrawn_key]
+        bool_keys = [consent_withdrawn_key]
+        equal_keys = ["uid"]
+        concat_keys = []
         matrix_keys = []
-
-        for plan in PipelineConfiguration.RQA_CODING_PLANS + PipelineConfiguration.FOLLOW_UP_CODING_PLANS:
-            show_matrix_keys = list()
-            for code in plan.code_scheme.codes:
-                show_matrix_keys.append(f"{plan.analysis_file_key}{code.string_value}")
-
-            AnalysisKeys.set_matrix_keys(
-                user, data, show_matrix_keys, plan.code_scheme, plan.coded_field, plan.analysis_file_key)
-
-            matrix_keys.extend(show_matrix_keys)
-
-        binary_keys = [plan.binary_analysis_file_key
-                       for plan in PipelineConfiguration.RQA_CODING_PLANS + PipelineConfiguration.FOLLOW_UP_CODING_PLANS
-                       if plan.binary_analysis_file_key is not None]
-
-        equal_keys = ["uid", "operator"]
-        equal_keys.extend(demog_keys)
-        concat_keys = [plan.raw_field for plan in PipelineConfiguration.RQA_CODING_PLANS]
-        follow_up_concat_keys = [plan.raw_field for plan in PipelineConfiguration.FOLLOW_UP_CODING_PLANS]
-        concat_keys.extend(follow_up_concat_keys)
-
-        bool_keys = [
-            consent_withdrawn_key,
-            "drama_promo",
```

Comment:
What happened to all the time keys?

Reply:
Re-RDA offline discussion they don't use them. The project has also accumulated the number of variables in analysis files columns, therefore, I think we better remove the unused ones.